tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -195,4 +199,33 @@ private function assertLimitResult(
             self::assertCount(count($expectedResults), $data);
         }
     }
+
+    public function testLimitWhenOrderByWithSubqueryWithOrderBy(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        if ($platform instanceof DB2Platform) {
+            self::markTestSkipped('Generated SQL is invalid');
+        }
+
+        $platformName = get_class($platform);
+        if (
+            $platformName === SQLServer2008Platform::class
+            || $platformName === SQLServer2005Platform::class
+        ) {
+            self::markTestSkipped('Generated SQL is invalid');
+        }
+
+        $this->connection->insert('modify_limit_table2', ['test_int' => 1]);
+        $this->connection->insert('modify_limit_table2', ['test_int' => 2]);
+
+        $subquery = $this->connection->createQueryBuilder()
+            ->select('test_int')
+            ->from('modify_limit_table2')
+            ->orderBy('test_int')
+            ->setMaxResults(1);
+
+        $sql = 'SELECT test_int FROM modify_limit_table2 ORDER BY (' . $subquery->getSQL() . ') ASC';
```

Comment:
What does this query try to achieve? It returns both rows in their original order. Why is this `ORDER BY` clause is needed? Additionally, it's different from the one in the unit test: the `LIMIT` clause is applied to the inner query which in turn doesn't have a sub-query which the original issue is above.

Reply:
I changed query and input so that ORDER BY means something.
I limit the query because I am getting the error that the subquery returned more than 1 row.